scarb/src/compiler/compilers/starknet_contract.rs

```diff
@@ -291,38 +305,53 @@ fn find_project_contracts(
     mut db: &dyn SemanticGroup,
     main_crate_ids: Vec<CrateId>,
     external_contracts: Option<Vec<ContractSelector>>,
-) -> Result<Vec<ContractDeclaration>> {
+) -> Result<Vec<ContractDeclaration>, anyhow::Error> {
     let internal_contracts = {
         let _ = trace_span!("find_internal_contracts").enter();
         find_contracts(db, &main_crate_ids)
     };
 
-    let external_contracts = if let Some(external_contracts) = external_contracts {
-        let _ = trace_span!("find_external_contracts").enter();
-        debug!("external contracts selectors: {:?}", external_contracts);
-
-        let crate_ids = external_contracts
-            .iter()
-            .map(|selector| selector.package().into())
-            .unique()
-            .map(|package_name: SmolStr| {
-                db.upcast_mut()
-                    .intern_crate(CrateLongId::Real(package_name))
-            })
-            .collect::<Vec<_>>();
-        find_contracts(db, crate_ids.as_ref())
-            .into_iter()
-            .filter(|decl| {
-                external_contracts.iter().any(|selector| {
+    let external_contracts: Vec<ContractDeclaration> =
+        if let Some(external_contracts) = external_contracts {
+            let _ = trace_span!("find_external_contracts").enter();
+            debug!("external contracts selectors: {:?}", external_contracts);
+
+            let crate_ids = external_contracts
+                .iter()
+                .map(|selector| selector.package().into())
+                .unique()
+                .map(|package_name: SmolStr| {
+                    db.upcast_mut()
+                        .intern_crate(CrateLongId::Real(package_name))
+                })
+                .collect::<Vec<_>>();
+            let contracts = find_contracts(db, crate_ids.as_ref());
+            let filtered_contracts: Result<Vec<_>, _> = contracts
+                .into_iter()
+                .filter_map(|decl| {
                     let contract_path = decl.module_id().full_path(db.upcast());
-                    contract_path == selector.full_path()
+                    let is_matched = external_contracts.iter().any(|selector| {
+                        if selector.is_wildcard() {
+                            match selector.partial_path() {
+                                Ok(partial_path) => contract_path.starts_with(&partial_path),
+                                Err(_) => false,
+                            }
+                        } else {
+                            contract_path == selector.full_path()
+                        }
+                    });
+                    if is_matched {
+                        Some(Ok::<ContractDeclaration, anyhow::Error>(decl))
+                    } else {
+                        None
+                    }
```

Comment:
I would rather show the parsing error to the user, so he can adjust his config appropriately. 

My suggestion would be, to validate the struct `Prop` which deserializes this value. 
https://github.com/software-mansion/scarb/blob/0db29e2b584b624a3f0fc4b8026d2aae3dfc0113/scarb/src/compiler/compilers/starknet_contract.rs#L186C9-L186C9
This way we could check all the selectors once,  and show the errors to the user. 

We can also have a simple test case with invalid selector to verify the validation.

Reply:
removed all this, now ensure is enforced on every path of the external-contracts in `compile`